File: 01-arrays_&_hashing/00217-Contains_Duplicate/00217-contains_duplicate.go
// 217: Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/

package main

import "fmt"

// SOLUTION
func containsDuplicate(nums []int) bool {
	m := make(map[int]bool)
	
	for _, n := range nums {
		if _, ok := m[n]; ok {
			return true
		} else {
			m[n] = true
		}
	}

	return false
}

func main() {
	// INPUT
	nums := []int{1,2,3,1}

	// OUTPUT
	result := containsDuplicate(nums)
	fmt.Println(result)
}

File: 01-arrays_&_hashing/00217-Contains_Duplicate/00217-contains_duplicate.py
# 217: Contains Duplicate
# https://leetcode.com/problems/contains-duplicate/

class Solution:
    # SOLUTION
    def containsDuplicate(self, nums: list[int]) -> bool:
        return len(nums) > len(set(nums))

if __name__ == "__main__":
    o = Solution()

    # INPUT
    nums: list[int] = [1,2,3,1]

    # OUTPUT
    result = o.containsDuplicate(nums)
    print(result)
